src/test/java/com/google/step/datamanager/DealManagerDatastoreTest.java

```diff
@@ -0,0 +1,155 @@
+
+  private final DealManagerDatastore dealManagerDatastore = new DealManagerDatastore(searchManager);
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testCreate_success() {
+    Deal deal =
+        dealManagerDatastore.createDeal(
+            DESCRIPTION_A, BLOBKEY_A, DATE_A, DATE_B, SOURCE_A, USER_ID_A, RESTAURANT_ID_A);
+    assertEquals(DESCRIPTION_A, deal.description);
+    assertEquals(BLOBKEY_A, deal.photoBlobkey);
+    assertEquals(DATE_A, deal.start);
+    assertEquals(DATE_B, deal.end);
+    assertEquals(SOURCE_A, deal.source);
+    assertEquals(USER_ID_A, deal.posterId);
+    assertEquals(RESTAURANT_ID_A, deal.restaurantId);
+  }
+
+  @Test
+  public void testRead_success() {
+    Deal createdDeal =
+        dealManagerDatastore.createDeal(
+            DESCRIPTION_A, BLOBKEY_A, DATE_A, DATE_B, SOURCE_A, USER_ID_A, RESTAURANT_ID_A);
+    Deal deal = dealManagerDatastore.readDeal(createdDeal.id);
+    assertEquals(DESCRIPTION_A, deal.description);
+    assertEquals(BLOBKEY_A, deal.photoBlobkey);
+    assertEquals(DATE_A, deal.start);
+    assertEquals(DATE_B, deal.end);
+    assertEquals(SOURCE_A, deal.source);
+    assertEquals(USER_ID_A, deal.posterId);
+    assertEquals(RESTAURANT_ID_A, deal.restaurantId);
+  }
+
+  @Test
+  public void testRead_invalidId_returnsNull() {
+    Deal deal = dealManagerDatastore.readDeal(DEAL_ID_A);
+    assertNull(deal);
+  }
+
+  @Test
+  public void testDelete() {
+    Deal createdDeal =
+        dealManagerDatastore.createDeal(
+            DESCRIPTION_A, BLOBKEY_A, DATE_A, DATE_B, SOURCE_A, USER_ID_A, RESTAURANT_ID_A);
+    dealManagerDatastore.deleteDeal(createdDeal.id);
+    Deal deal = dealManagerDatastore.readDeal(createdDeal.id);
+    assertNull(deal);
+  }
+
+  @Test
+  public void testUpdate_invalidId_returnsNull() {
+    Deal deal = new Deal(DEAL_ID_A, null, null, null, null, null, -1, -1);
+    Deal updatedDeal = dealManagerDatastore.updateDeal(deal);
+    assertNull(updatedDeal);
+  }
+
+  @Test
+  public void testUpdate_descriptionOnly() {
+    Deal createdDeal =
+        dealManagerDatastore.createDeal(
+            DESCRIPTION_A, BLOBKEY_A, DATE_A, DATE_B, SOURCE_A, USER_ID_A, RESTAURANT_ID_A);
+    Deal dealToUpdate = new Deal(createdDeal.id, DESCRIPTION_B, null, null, null, null, -1, -1);
+    Deal updatedDeal = dealManagerDatastore.updateDeal(dealToUpdate);
+
+    // only description should change, everything else should remain
+    assertEquals(DESCRIPTION_B, updatedDeal.description);
+    assertEquals(BLOBKEY_A, updatedDeal.photoBlobkey);
+    assertEquals(DATE_A, updatedDeal.start);
+    assertEquals(DATE_B, updatedDeal.end);
+    assertEquals(SOURCE_A, updatedDeal.source);
+    assertEquals(USER_ID_A, updatedDeal.posterId);
+    assertEquals(RESTAURANT_ID_A, updatedDeal.restaurantId);
+  }
+
+  @Test
```

Comment:
i think you can use it here as well like assertThat(updatedDeal, samePropertyValuesAs(createdDeal));